core/client/fs/src/main/java/alluxio/client/file/cache/CacheManagerOptions.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache;
+
+import alluxio.client.file.cache.evictor.CacheEvictorOptions;
+import alluxio.client.file.cache.store.PageStoreOptions;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+
+import java.util.List;
+
+/**
+ * Options for initiating local cache manager.
+ */
+public class CacheManagerOptions {
+  private boolean mAsyncRestoreEnabled;
+  private boolean mAsyncWriteEnabled;
+  private int mAsyncWriteThreads;
+  private boolean mQuotaEnabled;
+  private int mMaxEvictionRetries;
+  private long mPageSize;
+  private List<PageStoreOptions> mPageStoreOptions;
+  private CacheEvictorOptions mCacheEvictorOptions;
+
+  /**
+   * @param conf
+   * @return instance of CacheManagerOptions
+   */
+  public static CacheManagerOptions create(AlluxioConfiguration conf) {
+    CacheEvictorOptions cacheEvictorOptions = new CacheEvictorOptions()
+        .setEvictorClass(conf.getClass(PropertyKey.USER_CLIENT_CACHE_EVICTOR_CLASS))
+        .setIsNondeterministic(
+            conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_EVICTOR_NONDETERMINISTIC_ENABLED));
+    CacheManagerOptions options = new CacheManagerOptions()
+        .setAsyncRestoreEnabled(
+            conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_ASYNC_RESTORE_ENABLED))
+        .setAsyncWriteThreads(conf.getInt(PropertyKey.USER_CLIENT_CACHE_ASYNC_WRITE_THREADS))
+        .setIsAsyncWriteEnabled(
+            conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_ASYNC_WRITE_ENABLED))
+        .setMaxEvictionRetries(conf.getInt(PropertyKey.USER_CLIENT_CACHE_EVICTION_RETRIES))
+        .setPageSize(conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE))
+        .setQuotaEnabled(conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_QUOTA_ENABLED))
+        .setCacheEvictorOptions(cacheEvictorOptions)
+        .setPageStoreOptions(PageStoreOptions.create(conf));
+    return options;
+  }
+
+  /**
+   * @param conf
+   * @return instance of CacheManagerOptions
+   */
+  public static CacheManagerOptions createForWorker(AlluxioConfiguration conf) {
+    CacheEvictorOptions cacheEvictorOptions = new CacheEvictorOptions()
+        .setEvictorClass(conf.getClass(PropertyKey.WORKER_PAGE_STORE_EVICTOR_CLASS))
```

Comment:
lol just curious, is there a way to avoid using two sets of nearly the same property keys?

Reply:
hmmm, it might be difficult, because the worker's pagestore and local cache might be enabled at the same time, we need a separate property set for page store

Reply:
Some property key templates? at least on the Property key file, only one set of property key name with definitions?